src/main/kotlin/controllers/CacheChooserController.kt

```diff
@@ -281,53 +306,169 @@ class CacheChooserController(
     }
 
     private fun populateCachesList(
-        cacheListModel: FilteredListModel<String>,
+        cacheListModel: FilteredListModel<OpenRs2Cache>,
         cacheList: Component,
         listCachesPlaceholder: JLabel
     ) {
-        try {
-            val doc = Jsoup.connect(RUNESTATS_URL).get()
-            cacheListModel.backingList = doc.select("a")
-                .map { col -> col.attr("href") }
-                .filter { it.length > 10 } // get rid of ../ and ./types
-                .reversed()
-            cacheList.isVisible = true
-            listCachesPlaceholder.isVisible = false
-        } catch (e: Exception) {
-            e.printStackTrace()
-            listCachesPlaceholder.text += "\n\n${e.message}"
-            if (e is SSLHandshakeException) {
-                listCachesPlaceholder.text += "\n\nSSLHandshakeException is a known bug with certain Java versions, try updating."
+        Thread {
+            try {
+                val openRs2Api = OpenRs2Api()
+                val allCaches = openRs2Api.getCaches()
+                
+                // Filter by oldschool game and sort by timestamp (newest first)
+                val filteredCaches = allCaches
+                    .filter { it.game == "oldschool" }
+                    .sortedByDescending { cache ->
+                        cache.timestamp?.let {
+                            try {
+                                Instant.parse(it).toEpochMilli()
+                            } catch (e: Exception) {
+                                0L
+                            }
+                        } ?: 0L
+                    }
+                
+                SwingUtilities.invokeLater {
+                    if (filteredCaches.isEmpty()) {
+                        listCachesPlaceholder.text = "No Old School RuneScape caches found."
+                        listCachesPlaceholder.isVisible = true
+                        cacheList.isVisible = false
+                    } else {
+                        cacheListModel.backingList = filteredCaches
+                        cacheList.isVisible = true
+                        listCachesPlaceholder.isVisible = false
+                    }
+                }
+            } catch (e: Exception) {
+                e.printStackTrace()
+                SwingUtilities.invokeLater {
+                    listCachesPlaceholder.text = "Error loading caches: ${e.message}"
+                    if (e is SSLHandshakeException) {
+                        listCachesPlaceholder.text += "\n\nSSLHandshakeException is a known bug with certain Java versions, try updating."
+                    }
+                    listCachesPlaceholder.isVisible = true
+                    cacheList.isVisible = false
+                }
             }
-        }
+        }.start()
     }
 
     private fun downloadCache(
```

Comment:
this is getting fairly hefty; might be nicer if it had its own class at this point. also it's been a while since I touched RS2 stuff but is there a reason to remove runestats support entirely? are they cancelled? lol